tool/tctl/common/plugin/entraid.go

```diff
@@ -0,0 +1,401 @@
+	if err != nil {
+		return settings, trace.Wrap(err, "failed to read Client ID")
+	}
+
+	if p.install.entraID.accessGraph {
+		dataValidator := func(input string) bool {
+			settings.accessGraphCache, err = readTAGCache(input)
+			return err == nil
+		}
+		_, err = readData(os.Stdin, os.Stdout, "Enter the Access Graph Cache file location", dataValidator, "File does not exist or is invalid")
+		if err != nil {
+			return settings, trace.Wrap(err, "failed to read Access Graph Cache file")
+		}
+	}
+	return settings, nil
+}
+
+// InstallEntra is the entry point for the `tctl plugins install entraid` command.
+// This function guides users through an interactive setup process to configure EntraID integration,
+// directing them to execute a script in Azure Cloud Shell and provide the required configuration inputs.
+// The script creates an Azure EntraID Enterprise Application, enabling SAML logins in Teleport with
+// the following claims:
+// - givenname: user.givenname
+// - surname: user.surname
+// - emailaddress: user.mail
+// - name: user.userprincipalname
+// - groups: user.groups
+// Additionally, the script establishes a Trust Policy in the application to allow Teleport
+// to be recognized as a credential issuer when system credentials are not used.
+// If system credentials are present, the script will skip the Trust policy creation using
+// system credentials for EntraID authentication.
+// Finally, if no system credentials are in use, the script will set up an Azure OIDC integration
+// in Teleport and a Teleport plugin to synchronize access lists from EntraID to Teleport.
+func (p *PluginsCommand) InstallEntra(ctx context.Context, args installPluginArgs) error {
+	inputs := p.install
+
+	proxyPublicAddr, err := getProxyPublicAddr(ctx, args.authClient)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	settings, err := p.entraSetupGuide(proxyPublicAddr)
+	if err != nil {
+		if errors.Is(err, errCancel) {
+			return nil
+		}
+		return trace.Wrap(err)
+	}
+
+	var tagSyncSettings *types.PluginEntraIDAccessGraphSettings
+	if settings.accessGraphCache != nil {
+		tagSyncSettings = &types.PluginEntraIDAccessGraphSettings{
+			AppSsoSettingsCache: settings.accessGraphCache.AppSsoSettingsCache,
+		}
+	}
+
+	saml, err := types.NewSAMLConnector(inputs.entraID.authConnectorName, types.SAMLConnectorSpecV2{
+		AssertionConsumerService: strings.TrimRight(proxyPublicAddr, "/") + "/v1/webapi/saml/acs/" + inputs.entraID.authConnectorName,
+		AllowIDPInitiated:        true,
+		// AttributesToRoles is required, but Entra ID does not have a default group (like Okta's "Everyone"),
+		// so we add a dummy claim that will never be fulfilled with the default configuration instead,
+		// and expect the user to modify it per their requirements.
+		AttributesToRoles: []types.AttributeMapping{
+			{
+				Name:  "https://example.com/my_attribute",
+				Value: "my_value",
+				Roles: []string{"requester"},
+			},
```

Comment:
How would a user know they need to modify these? Should the command ask for claim name and value as a part of this setup flow?